java/arcs/core/storage/driver/Database.kt

```diff
@@ -158,6 +160,15 @@ object DatabaseDriverProvider : DriverProvider {
         this._factory = databaseFactory
         this.schemaLookup = schemaLookup
         DriverFactory.register(this)
+        CapabilitiesResolver.registerKeyCreator(
+            DATABASE_DRIVER_PROTOCOL,
+            Capabilities.Persistent,
+            { (arcId), entitySchemaHash -> DatabaseStorageKey(
+                arcId.toString(),
+                entitySchemaHash,
+                false
+            ) }
+        )
```

Comment:
Would be cleaner as:

```kotlin
CapabilitiesResolver.registerKeyCreator(
    DATABASE_DRIVER_PROTOCOL, 
    Capabilities.Persistent
) { storageKeyOptions, entitySchemaHash ->
    DatabaseStorageKey(storageKeyOptions.arcId.toString(), entitySchemaHash, false)
}
```

Notice: the lambda, when the last param in a function call, typically goes outside of the closing `)`. Also, the wrapping is a bit confusing, so I tweaked that. Finally, the destructuring of the StorageKeyOptions here can be brittle, and it's not clear to the reader what it's really getting.

A final question: are we really going to want every arc to have its own database?

Reply:
wow, this is a weird syntax!

I thought the `unique` parameter indicates a place in a database, not an entire database, doesn't it?